soh/soh/ShipUtils.cpp

```diff
@@ -96,3 +97,53 @@ extern "C" void* Ship_GetCharFontTexture(u8 character) {
 
     return (void*)fontTbl[adjustedChar];
 }
+
+static bool rand_init = false;
+static uint64_t state = 0;
+const uint64_t multiplier = 6364136223846793005ULL;
+const uint64_t increment = 11634580027462260723ULL;
+
+// Initialize with seed specified
+void ShipUtils::RandInit(uint64_t seed) {
+    rand_init = true;
+    state = seed;
+}
+
+uint32_t ShipUtils_next32() {
+    if (!rand_init) {
+        // No seed given, get a random number from device to seed
+#if !defined(__SWITCH__) && !defined(__WIIU__)
+        uint64_t seed = static_cast<uint64_t>(std::random_device{}());
+#else
+        uint64_t seed = static_cast<uint64_t>(std::hash<std::string>{}(std::to_string(rand())));
```

Comment:
```suggestion
        uint64_t seed = static_cast<uint64_t>(rand());
```
Beyond this PR's scope, but we should go ahead & rely on rand without the string-then-hash song & dance

Reply:
If this was on dev-copper instead of dev I'd say it's probably out of scope but since its on dev I might as well go ahead and make the bigger changes, so I'm thinking this is maybe not quite out of scope after all.